app/controllers/application_controller.rb

```diff
@@ -45,6 +45,21 @@ def user_not_authorized
   end
 
   def pundit_user
-    PunditContext.new(current_user, { current_project: @project, current_story: @story })
+    PunditContext.new(current_team, current_user, { current_project: @project, current_story: @story })
+  end
+
+  def current_team
+    raise ActiveRecord::RecordNotFound, 'Team not set' unless session[:current_team_slug]
+    @current_team ||= Team.not_archived.find_by_slug(session[:current_team_slug])
```

Comment:
the policy will return the relation projects.not_archived and so the application controller will get a ActiveRecord::NotFound and raise a 404 (users from an archived team will not be able to login, for example, unless they are member of non-archived teams as well, you must be enrolled to at least one active team). The first time a new user tries to sign up to a new team, it is automatically enrolled as the administrator for that team.